demo/src/main/java/com/example/demo/dao/user/LoadDatabase.java

```diff
@@ -29,11 +36,16 @@ CommandLineRunner initDatabase(UserRepository repository, GameStateRecordDao gam
             User testUser1 = repository.save(new User("Test 1", 101L));
             User testUser2 = repository.save(new User("Test 2", 202L));
 
-            GameStateRecord record = new GameStateRecord(List.of(testUser1.getId(), testUser2.getId()));
+            UserService user = new UserServiceImpl(repository);
```

Comment:
use spring injection, not manual object creation in this particular case. 
otherwise there will be several instances of UserService